src/operand/reference.ts

```diff
@@ -9,24 +9,35 @@ type Keys = (string | number)[]
 const keyWithArrayIndexRegex =
   /^(?<currentKey>[^[\]]+?)(?<indexes>(?:\[\d+])+)?$/
 const arrayIndexRegex = /\[(\d+)]/g
-const parseKey = (key: string): Keys =>
-  key.split('.').flatMap((key) => {
-    const parseResult = keyWithArrayIndexRegex.exec(key)
-    const keys: Keys = []
-    if (parseResult) {
-      keys.push(parseResult?.groups?.currentKey ?? key)
-
-      const rawIndexes = parseResult?.groups?.indexes
-      if (rawIndexes) {
-        for (const indexResult of rawIndexes.matchAll(arrayIndexRegex)) {
-          keys.push(parseInt(indexResult[1]))
+
+const parseBacktickWrappedKey = (key: string) =>
+  key.startsWith('`') && key.endsWith('`') ? key.slice(1, -1) : key
+
+const parseKey = (key: string): Keys => {
+  const keys = key.match(/(`[^[\]]+`(\[\d+\])*|[^`.]+)/g)
```

Comment:
What is this expected behavior if there is an odd-number of backticks?

Reply:
This regex allows the key to has backticks in it as long as it is wrapped with two backticks.
The only conflict is if there is a key that really starts and ends with a backtick. But I don't this would happen.